core/src/test/java/kafka/test/api/ProducerRebootstrapTest.java

```diff
@@ -0,0 +1,121 @@
+import org.apache.kafka.common.config.TopicConfig;
+import org.apache.kafka.common.test.ClusterInstance;
+import org.apache.kafka.common.test.api.ClusterConfig;
+import org.apache.kafka.common.test.api.ClusterTemplate;
+import org.apache.kafka.common.test.api.Type;
+import org.apache.kafka.coordinator.group.GroupCoordinatorConfig;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class ProducerRebootstrapTest {
+    private static final int BROKER_COUNT = 2;
+
+    static List<ClusterConfig> generator() {
+        Map<String, String> serverProperties = new HashMap<>();
+        // Enable unclean leader election for the test topic
+        serverProperties.put(TopicConfig.UNCLEAN_LEADER_ELECTION_ENABLE_CONFIG, "true");
+        serverProperties.put(GroupCoordinatorConfig.OFFSETS_TOPIC_REPLICATION_FACTOR_CONFIG, String.valueOf(BROKER_COUNT));
+
+        return Stream.of(false, true)
+                .map(ProducerRebootstrapTest::getRebootstrapConfig)
+                .map(rebootstrapProperties -> ProducerRebootstrapTest.buildConfig(serverProperties, rebootstrapProperties))
+                .toList();
+    }
+
+    static Map<String, String> getRebootstrapConfig(boolean useRebootstrapTriggerMs) {
+        Map<String, String> properties = new HashMap<>();
+        if (useRebootstrapTriggerMs) {
+            properties.put("metadata.recovery.rebootstrap.trigger.ms", "5000");
+        } else {
+            properties.put("metadata.recovery.rebootstrap.trigger.ms", "3600000");
+            properties.put("socket.connection.setup.timeout.ms", "5000");
+            properties.put("socket.connection.setup.timeout.max.ms", "5000");
+            properties.put("reconnect.backoff.ms", "1000");
+            properties.put("reconnect.backoff.max.ms", "1000");
+        }
+        properties.put("metadata.recovery.strategy", "rebootstrap");
+        properties.putIfAbsent(ProducerConfig.ACKS_CONFIG, "-1");
+        return properties;
+    }
+
+    static ClusterConfig buildConfig(Map<String, String> serverProperties, Map<String, String> rebootstrapProperties) {
+        return ClusterConfig.defaultBuilder()
+                .setTypes(Set.of(Type.KRAFT))
+                .setBrokers(BROKER_COUNT)
+                .setProducerProperties(rebootstrapProperties)
+                .setServerProperties(serverProperties).build();
+    }
+
+    @ClusterTemplate(value = "generator")
+    public void testRebootstrap(ClusterInstance clusterInstance) throws ExecutionException, InterruptedException {
+        String topic = "topic";
+        try (Admin admin = clusterInstance.admin()) {
+            admin.createTopics(Collections.singletonList(new NewTopic(topic, BROKER_COUNT, (short) 2))).all().get();
```

Comment:
Please use `List.of()` instead `Collections.singletonList`, and  I don't think we need to get the future value.
```
admin.createTopics(List.of(new NewTopic(topic, BROKER_COUNT, (short) 2)));
```